### ISSUE TITLE
How to replace playURIs with queueSpotifyURI

### ISSUE THREAD

**Issue by lorenzobrasini-spot** [spotify/ios-sdk#742]
Hello,
I was using `playURIs` to execute "anonymous" playlists and calling `skipNext` and `skipPrevious` always worked as expected.
Now I'm rewriting my code using `playSpotifyURI` and `queueSpotifyURI` but there's something different: I can play the first track and enqueue the second one, then it always returns an error:

> Error Domain=com.spotify.ios-sdk.playback Code=15 "An unknown error occurred." UserInfo=0x171a643c0 {NSLocalizedDescription=An unknown error occurred.

Also I can't navigate the playlist using `skipNext` and `skipPrevious`
Could someone please tell me how to do it with this new SDK? From the documentation I can't understand how to make it work.

Thanks
Lorenzo

**Comment by gchilds** [comment]
TL;DR `queueSpotifyURI` is sensitive to where it's called. I wrote some notes on it here: https://github.com/spotify/ios-sdk/issues/717#issuecomment-242882785

that's relative to Beta20, but makes sense with Beta21 if you replace `didChangePlaybackState ` with `didChangeMetadata` it still makes sense.

**Comment by lorenzobrasini-spot** [comment]
I'm calling `queueSpotifyURI` from `didReceivePlaybackEvent withName` but I can add only one track to the queue this way.
Also I can `skipNext` but I cant navigate back and forth my playlist because the queued tracks arn't available anymore once played.
I hope some more documentation/examples will be released from Spotify developers because now they're very poor.

L.

**Comment by Voxar** [comment]
Hi!

The error code translates to kSpErrorAPIRateLimited. The current rate seems to be 10 playback related api calls per second.
I'm not sure why it would be designed this way but try adding tracks slower :/

skipNext should skip to the next queued track but as the queue works the "consumed" tracks will be popped from the queue when playing them. To my knowledge, skipPrevious should return to the original track played tho. 

I'll add the missing error descriptions for the next version, thanks!


The underlying library was bumped far and it has been slimmed down a lot since the last version we used. Think of it being as thin as an audio player playing only the file you tell it to. 

Feel free to implement your own playlist code and progress playback based on callbacks from the player until we catch up with functionality.

Thanks

**Comment by gchilds** [comment]
@Voxar So you plan bring back playlists?

**Comment by Voxar** [comment]
@gchilds Support for arbitrary lists of tracks is not in our backlog at the moment.

**Comment by CVertex** [comment]
@Voxar @erdtman 
The Spotify web widget supports ad-hoc playlists. It would be consistent for the mobile SDKs to support them too.

They are very handy for designing for listeners to experiment with the order of songs before saving to a playlist. 

Is there a technical reason ad-hoc playlists are not supported?

**Comment by erdtman** [comment]
As mentioned by voxar the underlying library has been slimmed down and does not support this functionality.

I cannot do much about that but I will relay your which to the team responsible for that component.

**Comment by CVertex** [comment]
@erdtman thanks for your explanation.
We'd really appreciate you informing them. 
My feeling is that you should aim for some feature parity to match web widget, which is very flexible

**Comment by Voxar** [comment]
Thank you for your inputs

The removal of playURIs seems to stem from a product decision for a separate branch that happens to affect the mobile SDK's due to some shared code. I'll see what I can do.
